sync2/rangesync/minhash.go

```diff
@@ -0,0 +1,97 @@
+package rangesync
+
+import (
+	"cmp"
+	"fmt"
+	"slices"
+
+	"github.com/spacemeshos/go-scale"
+
+	"github.com/spacemeshos/go-spacemesh/sync2/types"
+)
+
+// MinhashSampleItem represents an item of minhash sample subset.
+type MinhashSampleItem uint32
+
+func (m MinhashSampleItem) String() string {
+	return fmt.Sprintf("0x%08x", uint32(m))
```

Comment:
will fix, thanks

Reply:
Well actually using `m` instead of `uint32(m)` causes a build-time error
```
./minhash.go:15:9: fmt.Sprintf format %08x with arg m causes recursive (github.com/spacemeshos/go-spacemesh/sync2/rangesync.MinhashSampleItem).String method call
```